Framework/Kernel/src/ErrorReporter.cpp

```diff
@@ -0,0 +1,141 @@
+#include "MantidKernel/ErrorReporter.h"
+#include "MantidKernel/ChecksumHelper.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidKernel/DateAndTime.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/InternetHelper.h"
+#include "MantidKernel/MantidVersion.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/ParaViewVersion.h"
+#include "MantidKernel/FacilityInfo.h"
+
+#include <Poco/ActiveResult.h>
+
+#include <json/json.h>
+
+namespace Mantid {
+namespace Kernel {
+
+namespace {
+/// static logger
+Logger g_log("ErrorReporter");
+}
+
+//----------------------------------------------------------------------------------------------
+// Constructor for ErrorReporter
+/** Constructor
+*/
+ErrorReporter::ErrorReporter(std::string application,
```

Comment:
We can use [delegating constructors](http://en.cppreference.com/w/cpp/language/initializer_list) (search for delegating constructors on the page) here to reduce the copy-and-pasted code:

```
ErrorReporter::ErrorReporter(std::string application,
                             Types::Core::time_duration upTime,
                             std::string exitCode, bool share) : ErrorReporter(application, upTime, exitCode, share, "", "")
```

This way we only define the constructor logic once along with the general case of all parameters and then simply use it elsewhere. This was only possible in >=C++11.